[noellethen] iP

noellethen opened the pull request.

noellethen committed:
Rename chatbot

noellethen committed:
Implement greet & exit

Commit:
Personalise chatbot style

Commit:
Update README.md

Commit:
Implement input echo

Echo user input with divider formatting. Add a personalised message upon initialisation and termination of chatbot when user types "bye".

This improves user experience.

Commit:
Add simple to-do list feature

Commit:
Add Task class

Commit:
Implement mark as done and unmark as done

Commit:
Add Todo subclass

Commit:
Add Deadline subclass

Commit:
Add Event subclass

Commit:
Refactor code

Moved some code that is more applicable to Task.java from Poodle.java to Task.java for readability.

Commit:
Add support for tracking ToDos, Events and Deadlines

Commit:
Edit texting style of chatbot

Commit:
Edit input.txt and EXPECTED.TXT for testing of Poodle

shennontay commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+    private static final String EVENT_COMMAND = "event";
+    private static final int SLASH_BY_LENGTH = 4;
+    private static final int SLASH_FROM_LENGTH = 6;
+    private static final int SLASH_TO_LENGTH = 4;
```
constants are clear and makes code more readable :)

shennontay replied:
constant names are clear and makes code more readable :)

shennontay commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+        try {
+            int firstSpaceIndex = input.indexOf(' ');
+            int taskNumber = Integer.parseInt(input.substring(firstSpaceIndex + 1));
+            if (taskNumber < 1 || taskNumber > Task.getTaskCount()) {
```
good checks to prevent out of bounds exceptions

shennontay replied:
good checks to prevent out of bounds exceptions

shennontay commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+            """;
+    private static final String EXIT_TEXT = """
+            --------------------------------------------
+            awwwww bye :c hope to see you again soon! <3
```
engaging and friendly messages, shows user-centric design thinking<3

shennontay replied:
engaging and friendly messages, shows user-centric design thinking<3

shennontay commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+                return;
+            }
+
+            Task t = Task.getTaskList()[taskNumber - 1];
```
taskNumber - 1 can be explained to be clearer

shennontay replied:
taskNumber - 1 can be explained to be clearer

Joannaj00 replied:
I think you could use a better name than "t"

shennontay reviewed (commented):
generally very readable and complies with coding conventions :))

Joannaj00 commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+public class Poodle {
+
+    // Text for greet and exit messages
+    private static final String DIVIDER = "--------------------------------------------";
```
good job making the constants all uppercase

Joannaj00 commented on src/main/java/Poodle.java:
```diff
@@ -0,0 +1,182 @@
+        case EVENT_COMMAND:
+            t = processEvent(input);
+            break;
+        default:
```
good structure for the switch statement!!

Joannaj00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public static Task[] getTaskList() {
+        return taskList;
```
good job making the method names verbs and also camelCase!!

noellethen committed:
Add error handling for some commands

Implemented error handling for todo, deadline and event having no description.
Implemented error handling for user input not corresponding to any commands.

noellethen committed:
Add PoodleException class to represent exceptions specific to Poodle

noellethen committed:
Refactor Poodle class to delegate exception handling to PoodleException

noellethen committed:
Update error handling for outOfRangeException

noellethen committed:
Merge pull request #1 from noellethen/branch-Level-5

branch-Level-5

noellethen committed:
Organise classes into packages

noellethen committed:
Merge pull request #2 from noellethen/branch-A-Packages

Organise classes into packages

noellethen committed:
Fix README.md

Removed "ui." in "ui.Poodle project template".

noellethen committed:
Edit Task.java to use ArrayList

noellethen committed:
Add support for deleting tasks from the list

noellethen committed:
Add Storage class to handle saving and loading tasks

noellethen committed:
Refactor Poodle.java to accommodate saving and loading tasks through Storage class

noellethen committed:
Merge pull request #3 from noellethen/branch-Level-6

Branch level 6

noellethen committed:
Merge branch 'master' into branch-Level-7

noellethen committed:
Merge pull request #4 from noellethen/branch-Level-7

Branch level 7

noellethen committed:
Fix bug where event tasks are saved incorrectly

noellethen committed:
Add Printer class to handle display of output

noellethen committed:
Bug fix where chatbot wrongly outputs for exit command

noellethen committed:
Fix error messages

syCHEN1645 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+package task;
+
+public class Deadline extends Task {
+    protected String by;
```
Maybe consider using nouns as string names?

syCHEN1645 commented on src/main/java/ui/Poodle.java:
```diff
@@ -0,0 +1,226 @@
+
+public class Poodle {
+    // Commands
+    private static final String LIST_COMMAND = "list";
```
Good job refactoring string literals!

syCHEN1645 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,92 @@
+            throw PoodleException.fileError("uhhhh i can't find the file: " + e.getMessage());
+        }
+    }
+}
```
This method looks a bit long, maybe consider breaking it down further?

syCHEN1645 commented on src/main/java/ui/Poodle.java:
```diff
@@ -0,0 +1,226 @@
+        }
+
+        return new Deadline(description, by);
+
```
Be careful with unnecessary empty lines.

syCHEN1645 commented on src/main/java/ui/Poodle.java:
```diff
@@ -0,0 +1,226 @@
+        int toIndex = input.indexOf("/to");
+        if (fromIndex == -1 || toIndex == -1 || toIndex < fromIndex) {
+            throw PoodleException.wrongFormatException(
+                    "type something like this: event <description> /from <start> /to <end>!");
```
Good informative messages!

syCHEN1645 reviewed (commented):
Nice and tidy code, keep it up!

noellethen committed:
Rename Printer.java to Ui.java

noellethen committed:
Add Parser.java to deal with making sense of user command

noellethen committed:
Add TaskList.java to deal with task modification operations

noellethen committed:
Refactor Poodle.java to accommodate previous abstractions

noellethen committed:
Add findTasks method to allow users to find tasks by keyword

noellethen committed:
Refactor existing classes to accommodate findTasks feature

noellethen committed:
Merge pull request #5 from noellethen/branch-Level-9

Branch level 9

noellethen committed:
Add JavaDoc comments

noellethen committed:
Merge pull request #6 from noellethen/branch-A-JavaDoc

Add JavaDoc comments

noellethen committed:
Update user guide

noellethen committed:
Edit user guide

noellethen committed:
Fix Storage.java to properly handle creating new .txt file when unable to locate it
